web/packages/shared/components/FileTransfer/FileTransfer.tsx

```diff
@@ -95,13 +106,13 @@ export function FileTransfer(props: FileTransferProps) {
 export function FileTransferDialog(
   props: Pick<
     FileTransferProps,
-    'transferHandlers' | 'backgroundColor' | 'errorText'
+    'transferHandlers' | 'backgroundColor' | 'errorText' | 'filesStore'
   > & {
     openedDialog: FileTransferDialogDirection;
     onCloseDialog(isAnyTransferInProgress: boolean): void;
   }
 ) {
-  const filesStore = useFilesStore();
```

Comment:
I don't like how `useFilesStore` was uprooted from here. As far as I can tell, this was done because `useSshSession` needs to be able to call `filesStore.start`. Uprooting it from here has a couple of consequences:

* Each callsite which used to use `FileTransfer` now needs to call `useFilesStore`, even though it's irrelevant to the callsite and the callsite does it only in order to pass it further down the line.
* `useSshSession` only needs `filesStore.start`, but it ends up passing the `filesStore` three levels up (`useFilesStore` -> `useFileTransfer` -> `useSshSession`) and then back down to `FileTransfer`.

There's probably a couple of ways to address this:

* Would it be feasible to move the functionality of `useFilesStore` within `useFileTransferContext`?
   * If so, `FileTransfer` could just call `useFileTransferContext`, the callsite wouldn't need to call a hook only to pass down the result.
   * `useSshSession` could get the `start` function from the context instead of having to call `useFilesStore` itself. `DocumentSsh` would need to be wrapped in `FileTransferContextProvider` instead of mounting this component itself but I think that's doable.
*  If `useSshSession` needs only `filesStore.start`, couldn't it receive that function as an argument? Nesting hooks like this seems like roundabout way to achieve this if reusing shared logic is the goal.
   * Even if `useFilesStore` was moved into a context, it'd probably still makes sense to accept `start` as an argument to make the relationship explicit and avoid the requirement of mounting file transfer context before a callsite is able to call `useSshSession`.

@gzdunek Could you help us out here? I think you've done some work in this area.

Reply:
Sounds good, will update shortly. thanks!